cmake: modules: shields: optional pre_dt_shield.cmake file

PaulSchaetzle opened the pull request:
I have a shield that uses a I2C mux which has multiple devices with the same I2C-address connected to it. This generates devicetree warnings while building because of the duplicated addresses. Boards have the pre_dt_board.cmake file which can be used to get around this issue (and more of course), however shields lack the capability to run any cmake code before the devicetree phase. The proposed change simply includes a file called pre_dt_shield.cmake, if it exists, for each active shield.

PaulSchaetzle committed:
cmake: modules: shields: pre_dt_shield.cmake file

Similar to the pre_dt_board.cmake, this is an optional file for
additional arguments that need to get passed to the devicetree compiler.

Signed-off-by: Paul Schaetzle <paulschaetzle@mailbox.org>

pdgendt commented:
Some comments:
- Update the commit subject to `cmake: modules: shields: optional pre_dt_cmake file`
- Add an entry to the documentation at `doc/build/dts/intro-input-output.rst`

PaulSchaetzle committed:
doc: dts: explain the usage of pre_dt_shield.cmake

Mention the possible use of pre_dt_shield.cmake in the devicetree and
shield documentation.

Signed-off-by: Paul Schaetzle <paulschaetzle@mailbox.org>

pdgendt commented:
> Some comments:
> 
> * Update the commit subject to `cmake: modules: shields: optional pre_dt_shield.cmake file`

I meant the git commit subject :)

> * Add an entry to the documentation at `doc/build/dts/intro-input-output.rst`

Thanks for the doc addition.
